src/compute/src/render/context.rs

```diff
@@ -1064,44 +1069,48 @@ where
     fn specialized_arrange(
         name: &String,
         oks: Collection<S, Row, i64>,
-        key: &Vec<MirScalarExpr>,
-        thinning: &Vec<usize>,
+        key: Vec<MirScalarExpr>,
+        thinning: Vec<usize>,
     ) -> (MzArrangement<S>, Collection<S, DataflowError, i64>) {
-        // Catch-all: Just use RowRow.
-        let (oks, errs) = oks
-            .map_fallible::<CapacityContainerBuilder<_>, CapacityContainerBuilder<_>, _, _, _>(
-                "FormArrangementKey",
-                specialized_arrangement_key(key.clone(), thinning.clone()),
+        let (oks, errs) =
+            oks.inner
+                .unary_fallible::<ColumnBuilder<((Row, Row), S::Timestamp, i64)>, _, _, _>(
+                    Pipeline,
+                    "FormArrangementKey",
+                    move |_, _| {
+                        Box::new(move |input, ok, err| {
+                            let mut key_buf = Row::default();
+                            let mut val_buf = Row::default();
+                            let mut datums = DatumVec::new();
+                            let temp_storage = RowArena::new();
+                            while let Some((time, data)) = input.next() {
+                                let mut ok_session = ok.session_with_builder(&time);
+                                let mut err_session = err.session(&time);
+                                for (row, time, diff) in data.iter() {
+                                    temp_storage.clear();
+                                    let datums = datums.borrow_with(row);
+                                    let val_datum_iter = thinning.iter().map(|c| datums[*c]);
```

Comment:
Bit of a nit, but can `val_datum_iter` be moved into the `Ok(())` branch? Mostly thinking about code clarity, and that one doesn't need to understand this term heading in to the `match` statement.

Reply:
Fixed.